### PR TITLE
Improve logging as we filter jobs from being considered.

### PR DESCRIPTION
**Pull request by scrosby**
## Changes proposed in this PR

- Log as we filter out jobs from being considered

## Why are we making these changes?
- Make it easier to determine where and why we cease to consider jobs for scheduling

### CODE REVIEW

File: scheduler/src/cook/scheduler/scheduler.clj [hunk 1]
```diff
@@ -582,18 +582,24 @@
...
-             (filter (fn [job] (util/job-allowed-to-start? db job)))
-             (filter user-within-launch-rate-limit?-fn)
-             (filter launch-plugin/filter-job-launches)
-             (take num-considerable)
-             ; Force this to be taken eagerly so that the log line is accurate.
-             (doall))]
+        quota-filter-jobs (util/filter-based-on-quota user->quota user->usage pending-jobs)
+        allowed-to-start-jobs (filter (fn [job] (util/job-allowed-to-start? db job)) quota-filter-jobs)
+        within-launch-rate-jobs (filter user-within-launch-rate-limit?-fn allowed-to-start-jobs)
+        launch-plugin-jobs (filter launch-plugin/filter-job-launches within-launch-rate-jobs)
```
Comment (dposada) [inline scheduler/src/cook/scheduler/scheduler.clj]: What does this filter do? The name `launch-plugin-jobs` doesn't tell the reader anything about what's happening.

File: scheduler/src/cook/scheduler/scheduler.clj [hunk 2]
```diff
@@ -582,18 +582,24 @@
...
+        considerable-jobs (take num-considerable launch-plugin-jobs)
+        ; Force this to be taken eagerly so that the log line is accurate.
+        considerable-jobs* (doall considerable-jobs)]
     (swap! pool->user->number-jobs update pool-name (constantly @user->number-jobs))
     (log/info "Users whose job launches are rate-limited " @user->rate-limit-count "( enforcing =" enforcing-job-launch-rate-limit? ")")
-    considerable-jobs))
+    (log/info "In" pool-name "pool, there are"
+              (count pending-jobs) " pending ->"
+              (count quota-filter-jobs) " quota-filtered ->"
+              (count allowed-to-start-jobs) " allowed-start ->"
```
Comment (dposada) [inline scheduler/src/cook/scheduler/scheduler.clj]:
```suggestion
              (count allowed-to-start-jobs) " allowed-to-start ->"
```

File: scheduler/src/cook/scheduler/scheduler.clj [hunk 3]
```diff
@@ -582,18 +582,24 @@
...
+              (count within-launch-rate-jobs) " launch-rate ->"
```
Comment (dposada) [inline scheduler/src/cook/scheduler/scheduler.clj]:
```suggestion
              (count within-launch-rate-jobs) " within-launch-rate ->"
```

File: scheduler/src/cook/scheduler/scheduler.clj [hunk 4]
```diff
@@ -582,18 +582,24 @@
...
+              (count launch-plugin-jobs) " launch-plugin ->"
```
Comment (dposada) [inline scheduler/src/cook/scheduler/scheduler.clj]: What does "launch-plugin" mean? We should make this be more obvious to someone reading the log line.

File: scheduler/src/cook/scheduler/scheduler.clj [hunk 5]
```diff
@@ -582,18 +582,24 @@
...
+              "and"
+              (count considerable-jobs) "considerable jobs.")
```
Comment (dposada) [inline scheduler/src/cook/scheduler/scheduler.clj]: Should this be `considerable-jobs*`?

Comment (dposada) [inline scheduler/src/cook/scheduler/scheduler.clj]: Let's remove the period from the end of the log line

Comment (scrosby) [reply scheduler/src/cook/scheduler/scheduler.clj]: Good catch.

Comment (scrosby) [reply scheduler/src/cook/scheduler/scheduler.clj]: Ok

File: scheduler/src/cook/scheduler/scheduler.clj (hunk 1 above)
Comment (dposada) [reply scheduler/src/cook/scheduler/scheduler.clj]: If we're going to give it a name, let's give it a meaningful name

File: scheduler/src/cook/scheduler/scheduler.clj (hunk 4 above)
Comment (dposada) [reply scheduler/src/cook/scheduler/scheduler.clj]: What do you mean by "any reason at all"? I consider myself an experienced developer, but I don't know what this means. All of the other parts of the log line make sense.

File: scheduler/src/cook/scheduler/scheduler.clj [hunk 6]
```diff
@@ -582,18 +582,24 @@
...
     (swap! pool->user->number-jobs update pool-name (constantly @user->number-jobs))
     (log/info "Users whose job launches are rate-limited " @user->rate-limit-count "( enforcing =" enforcing-job-launch-rate-limit? ")")
-    considerable-jobs))
+    (log/info "In" pool-name "pool, there are"
+              (count pending-jobs) "pending ->"
+              (count quota-filter-jobs) "quota-filtered ->"
+              (count allowed-to-start-jobs) "allowed-to-start ->"
+              (count within-launch-rate-jobs) "within-launch-rate ->"
+              (count launch-plugin-jobs) "allowed by launch-plugin ->"
+              " and"
```
Comment (dposada) [inline scheduler/src/cook/scheduler/scheduler.clj]:
```suggestion
              "and"
```

File: scheduler/src/cook/scheduler/scheduler.clj [hunk 7]
```diff
@@ -582,18 +582,24 @@
...
+        considerable-jobs* (doall considerable-jobs)]
     (swap! pool->user->number-jobs update pool-name (constantly @user->number-jobs))
     (log/info "Users whose job launches are rate-limited " @user->rate-limit-count "( enforcing =" enforcing-job-launch-rate-limit? ")")
-    considerable-jobs))
+    (log/info "In" pool-name "pool, there are"
+              (count pending-jobs) " pending ->"
+              (count quota-filter-jobs) " quota-filtered ->"
+              (count allowed-to-start-jobs) " allowed-to-start ->"
+              (count within-launch-rate-jobs) " within-launch-rate ->"
+              (count launch-plugin-jobs) " allowed by launch-plugin ->"
```
[the remark this answers is not in the document]
Comment (scrosby) [reply scheduler/src/cook/scheduler/scheduler.clj]: Changed.

**Comment by dposada** [comment]
Will merge when green

**Pull request merged**